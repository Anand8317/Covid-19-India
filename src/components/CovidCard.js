import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FiArrowRightCircle } from 'react-icons/fi';
import covidLogo from '../Images/covidImg.png';

const CovidCard = ({
  title, text, className,
}) => (
  <Link to={`/Covid/${title}`} key={title} className={className}>
    <div>
      <img src={covidLogo} alt="covidLogo" className="covid-img" />
    </div>
    <div className="subgrid">
      <div className="arrow-container">
        <FiArrowRightCircle className="arrow" />
      </div>
      <h3>{title}</h3>
      <p>{text}</p>
    </div>
  </Link>
);

CovidCard.propTypes = {
  title: PropTypes.string,
  text: PropTypes.number,
  className: PropTypes.string,
};

CovidCard.defaultProps = {
  title: '',
  text: 1,
  className: '',
};

export default CovidCard;
