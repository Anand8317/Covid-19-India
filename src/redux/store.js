import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import { covidReducer } from './covid/covid';

const store = configureStore(
  {
    reducer: {
      covid: covidReducer,
    },
  },
  applyMiddleware(thunk),
);
export default store;
