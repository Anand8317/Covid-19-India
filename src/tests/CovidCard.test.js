import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CovidCard from '../components/Covid';
import store from '../redux/store';

const App = () => {
  <Provider store={store}>
    <CovidCard />
  </Provider>;
};

describe('CovidCard component test', () => {
  it('Testing covid card page rendering', () => {
    const CovidCardComponent = renderer.create(<App />).toJSON();
    expect(CovidCardComponent).toMatchSnapshot();
  });
});
