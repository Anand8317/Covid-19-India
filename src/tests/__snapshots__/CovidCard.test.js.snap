// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Covid component test Testing covid page rendering 1`] = `null`;

exports[`CovidCard component test Testing covid card page rendering 1`] = `null`;
